### PR TITLE
[BEAM-8676] sdks/java: gax and grpc upgrades

### PR DESCRIPTION
**Pull request by suztomo**
https://issues.apache.org/jira/browse/BEAM-8676

Gax and gRPC work together. Upgrading their dependencies to higher versions.

- The latest gax-grpc:1.52.0 depends on gRPC 1.25.0. (So not setting grpc 1.26.0)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PostCommit

File: sdks/java/io/google-cloud-platform/build.gradle [hunk 1]
```diff
@@ -33,6 +33,7 @@ dependencies {
   compile project(":sdks:java:extensions:protobuf")
   compile library.java.avro
   compile library.java.bigdataoss_util
+  compile library.java.gax
```
Comment (suztomo) [inline sdks/java/io/google-cloud-platform/build.gradle]: As BigQueryServicesImpl uses gax's `com.google.api.gax.core.FixedCredentialsProvider` (undeclared used dependency), declaring it explicitly.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -372,7 +373,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_cloud_bigdataoss_version = "1.9.16"
     def google_cloud_core_version = "1.61.0"
     def google_cloud_spanner_version = "1.6.0"
-    def grpc_version = "1.17.1"
+    def grpc_version = "1.25.0"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: The latest version of gRPC is 1.26.0, but setting 1.25.0 as gax-grpc:1.52.0 depends on 1.25.0.

**Comment by suztomo** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by suztomo** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Python PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
R: @kennknowles

**Comment by suztomo** [comment]
R: @lukecwik @chamikaramj

**Comment by suztomo** [comment]
Run Java Precommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by suztomo** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run CommunityMetrics PreCommit

**Comment by suztomo** [comment]
retest this please

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by suztomo** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
retest this please

**Comment by suztomo** [comment]
retest this please

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by lukecwik** [comment]
Please post the before and after results of the [linkage checker](https://github.com/apache/beam/blob/0851849dcfe9b8088d69a5bdd3b55f7806b83ea9/build.gradle#L290) for the GCP IO module.

Swapping to gRPC 1.25 suggests to update [protobuf to 3.11.0 and guava to 28.1](https://mvnrepository.com/artifact/io.grpc/grpc-protobuf/1.26.0). Upgrading those may need to happen before this change could be adopted depending on the linkage checker results.

This seems like an innocuous change but past experience has shown that grpc/proto/guava like to make breaking changes.

**Comment by suztomo** [comment]
> Swapping to gRPC 1.25 suggests to update protobuf to 3.11.0 and guava to 28.1

Good point. The current protobuf 3.6.0 is not compatible with gax-grpc 1.52.0. Let me update that first. I'll attach linkage check result on that PR.

**Comment by suztomo** [comment]
Raised https://github.com/apache/beam/pull/10498 for protobuf upgrade. Jenkins is not running tests (yet?).

**Comment by suztomo** [comment]
```
Linkage Error on com.google.auth:google-auth-library-oauth2-http:0.12.0 x com.google.api:gax:1.52.0
(com.google.auth:google-auth-library-oauth2-http:0.12.0) com.google.auth.oauth2.ServiceAccountCredentials's method getQuotaProjectId() is not found (type: SYMBOL_NOT_FOUND)
com.google.api.gax.core.GoogleCredentialsProvider (com.google.api:gax:1.52.0)
(com.google.auth:google-auth-library-oauth2-http:0.12.0) com.google.auth.oauth2.ServiceAccountJwtAccessCredentials$Builder's method setQuotaProjectId(String arg1) is not found (type: SYMBOL_NOT_FOUND)
com.google.api.gax.core.GoogleCredentialsProvider (com.google.api:gax:1.52.0)
```

This PR needs https://github.com/apache/beam/pull/10508 first.

**Pull request closed without merging**